cognify/optimizer/evaluator.py

```diff
@@ -307,13 +310,14 @@ def evaluate_program(
         _be_quiet()
         # directly raise interrupt signal
         _stay_alert()
-        
+
+        sema.acquire()
```

Comment:
what's this for?

Reply:
Each thread is accessing a shared queue where we put the results of running a data point on the workflow.